src/ansible_navigator/steps.py

```diff
@@ -122,6 +128,90 @@ def _value_check(value: Any, want: Union[type, Tuple[type, ...]]) -> None:
             raise ValueError(f"wanted {want}, got {type(value)}")
 
 
+TValueType = TypeVar("TValueType")
+
+
+@dataclass
+class TypedStep(Generic[TValueType]):
+    # pylint: disable=too-many-instance-attributes
+    """One step in the flow of things.
+
+    This should be used for new actions following the pattern of a
+    TypedDict for the user interface as seen in settings.
```

Comment:
Looks like this is maybe
```suggestion
    :class:`~TypedDict` for the user interface as seen in settings.
```

Reply:
another mistake during iteration:

```This should be used for new actions following the pattern of  dataclasses for the user interface as seen in settings.```

or something like that